pkg/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/eec.go

```diff
@@ -0,0 +1,82 @@
+package modifiers
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/internal/statefulset/builder"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/extension"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/container"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ volumeModifier = EecModifier{}
+var _ volumeMountModifier = EecModifier{}
+var _ builder.Modifier = EecModifier{}
+
+const (
+	eecVolumeName = "eec-token"
+	eecMountPath  = "/var/lib/dynatrace/secrets/eec"
+	eecFile       = "token/eec.token"
+)
+
+func NewEecVolumeModifier(dk dynakube.DynaKube) EecModifier {
+	return EecModifier{
+		dk: dk,
+	}
+}
+
+type EecModifier struct {
+	dk dynakube.DynaKube
+}
+
+func (mod EecModifier) Enabled() bool {
+	return mod.dk.PrometheusEnabled()
+}
+
+func (mod EecModifier) Modify(sts *appsv1.StatefulSet) error {
+	baseContainer := container.FindContainerInPodSpec(&sts.Spec.Template.Spec, consts.ActiveGateContainerName)
+	sts.Spec.Template.Spec.Volumes = append(sts.Spec.Template.Spec.Volumes, mod.getVolumes()...)
+	baseContainer.VolumeMounts = append(baseContainer.VolumeMounts, mod.getVolumeMounts()...)
+
+	baseContainer.Command = []string{"/bin/sh"}
```

Comment:
please remove 43-46 - it's a workaround, it's not needed because AG has been fixed

Reply:
[Fixed here](https://github.com/Dynatrace/dynatrace-operator/pull/3497/commits/0c1242a98022fcb15b295175ab666b393fe79520)